### PR TITLE
NET-6788 - Mesh Gateway Host Port Support

### PR DESCRIPTION
**Pull request by sarahalsmiller**
### Changes proposed in this PR ###  
- Added hostPort support

### How I've tested this PR ###
- Ran local consul test

### Checklist ###
- [ ] Tests added
- [ ] [CHANGELOG entry added](https://github.com/hashicorp/consul-k8s/blob/main/CONTRIBUTING.md#adding-a-changelog-entry)

### CODE REVIEW

File: control-plane/gateways/deployment_dataplane_container.go [hunk 1]
```diff
@@ -112,11 +109,10 @@ func consulDataplaneContainer(config GatewayConfig, containerConfig *v2beta1.Gat
 	wanPort := corev1.ContainerPort{
 		Name:          "wan",
 		ContainerPort: int32(constants.DefaultWANPort),
+		HostPort:      containerConfig.HostPort,
 	}
 
-	if containerConfig != nil {
-		wanPort.ContainerPort = int32(443 + containerConfig.PortModifier)
-	}
+	wanPort.ContainerPort = 443 + containerConfig.PortModifier
```
Comment (NiniOak) [inline control-plane/gateways/deployment_dataplane_container.go:115]: Just ranting, why is hostPort int32 and containerPort just int?

File: control-plane/gateways/deployment_dataplane_container.go [hunk 2]
```diff
@@ -112,11 +109,10 @@ func consulDataplaneContainer(config GatewayConfig, containerConfig *v2beta1.Gat
 	wanPort := corev1.ContainerPort{
 		Name:          "wan",
 		ContainerPort: int32(constants.DefaultWANPort),
+		HostPort:      containerConfig.HostPort,
```
Comment (NiniOak) [inline control-plane/gateways/deployment_dataplane_container.go:112]: Thinking out loud, should we potentially set a default value for `hostPort` if we happen to run into an issue where a host port isn't exposed? Super hypothetical

File: control-plane/gateways/deployment_dataplane_container.go (hunk 1 above)
Comment (sarahalsmiller) [reply control-plane/gateways/deployment_dataplane_container.go:115]: It is int32, I updated the value to match what the object is expecting. https://github.com/hashicorp/consul-k8s/pull/3381/files#diff-3a9113026018fc0f58d311cb39bf0de995570740f1ebde6fa3cb1dcbb4e2c64bR101

File: control-plane/gateways/deployment_dataplane_container.go (hunk 2 above)
Comment (sarahalsmiller) [reply control-plane/gateways/deployment_dataplane_container.go:112]: I believe the defaultValue of host port as it exists today is 0, in which case the port won't be exposed. I could explicitly set that, but we get that behavior baked in when we initialize an empty struct.

File: control-plane/gateways/deployment_dataplane_container.go (hunk 1 above)
Comment (NiniOak) [reply control-plane/gateways/deployment_dataplane_container.go:115]: nice!

**Review by NiniOak — APPROVED** [review]

**Pull request merged**